Check kube-proxy health on linux

uthark opened the pull request:
Right now healthchecker supports kube-proxy on windows only.

This PR adds support for linux.

Other changes:
- refactored code for better reuse
- log if component is not supported.

uthark commented:
/assign @andyxning

uthark commented:
/assign @Random-Liu 
/assign @xueweiz

jeremyje commented:
Thanks for the PR. Seems fine I'd optionally suggest that there's a test/restructure added to ensure that Windows and Linux are supported moving forward but perhaps a bug is sufficient. @Random-Liu 

/lgtm
/ok-to-test

uthark commented:
@Random-Liu could you please take a look?

uthark commented:
@jeremyje @Random-Liu 

I moved `getHealthCheckFunc` to common code and added test to confirm it returns non-nil function for the supported components. 

Not sure what is the best way to abstract path to docker file. For CRI it is passed with CLI args, but docker/docker.exe were hardcoded. Do you want it to be configurable too?

jeremyje commented on pkg/healthchecker/health_checker.go:
```diff
@@ -110,3 +112,64 @@ func logPatternHealthCheck(service string, logPatternsToCheck map[string]int) (b
+}
+
+func getDockerPath() string {
+	if runtime.GOOS == "windows" {
```
Given that there's already a _windows.go and _linux.go file just define the methods there.

This has 2 benefits:
1. It's clear that when a new OS is supported that this method needs to be considered for porting.
2. The logic between OSes have lower risk of leaking into each other's binaries.

Random-Liu replied:
+1

Random-Liu commented on pkg/healthchecker/health_checker_test.go:
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package healthchecker
 
 import (
+	"k8s.io/node-problem-detector/cmd/healthchecker/options"
```
nit: Put this into the import group below

Random-Liu reviewed (commented):
LGTM, please address the comments.

Random-Liu commented:
And thanks for working on this! This is very useful feature. :)

uthark commented:
/retest

jeremyje commented:
/lgtm

It's possible the failures are coming about due to the kube-proxy checks are failing. The easiest way to verify is to create a Kubernetes cluster and swap out the NPD binary and see how it responds.
`kubectl get node [NODE_NAME] -o yaml` and look at the conditions.

jeremyje commented:
/lgtm

jeremyje commented:
/retest

Random-Liu commented:
/lgtm
/approve
/retest

Random-Liu commented:
/retest
